File: Day-06/Implement linked list.cpp
#include<iostream>
using namespace std;

class node {
    public:
        int val;
        node* next;
        node(int num) {
            val = num;
            next = NULL;
        }
};

node* insertNode(node* head,int val) {
    node* newNode = new node(val);
    
    if(head == NULL) {
        head = newNode;
        return head;
    }
    
    node* temp = head;
    while(temp->next != NULL) temp = temp->next;
    
    temp->next = newNode;
    return head;
}

void findMiddle(node* head)
{
  node* slow = head;
  node* fast = head;

  while(fast && fast->next)
  {
    slow = slow->next;
    fast = fast->next->next;
  }
  cout<<slow->val;
  
}

int main() {
    node* head = NULL;
    head = insertNode(head,2);
    head = insertNode(head,7);
    head = insertNode(head,11);
    head = insertNode(head,15);
    head = insertNode(head,21);
    findMiddle(head);
  
    return 0;
}

File: Day-06/Implement linked list.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

def insertNode(head, val):
    newNode = Node(val)
    if head == None:
        head = newNode
        return head
    temp = head
    while temp.next != None:
        temp = temp.next
    temp.next = newNode
    return head

def findMiddle(head: Node):
  slow = head
  fast = head
  while fast and fast.next:
    slow = slow.next
    fast = fast.next.next
  print(slow.val)


if __name__ == "__main__":
    head = None
    head = insertNode(head, 2)
    head = insertNode(head, 7)
    head = insertNode(head, 11)
    head = insertNode(head, 15)
    findMiddle(head)